src/CachedResponseClient.php

```diff
@@ -60,7 +60,11 @@ public function sendRequest(RequestInterface $request)
             $url = str_replace($this->apiKey, '[apikey]', $url);
         }
 
-        $file = sprintf('%s/%s_%s', $this->cacheDir, $host, sha1($url));
+        $cacheKey = $url;
+        if ('POST' === $request->getMethod()) {
+            $cacheKey .= $request->getBody();
```

Comment:
What if

```php

$cacheKey .= substr((string) $request->getBody(), 0, 1000) . $request->getBody()->getSize();
```

Reply:
hm.. or maybe it will not be an issue...
Also this is "just tests"

Reply:
What if two requests with large bodies differ only in the last few bytes? Should they not be cached as 2 separate requests?

Yes, I agree that this is a non issue. Better just leave it as it is.